#  Building a model.

# AdaBoost:

In [99]:
from sklearn.metrics import accuracy_score
accuracy_test_ada = accuracy_score(y_test,predict_test)
print("AdaBoost Classifier Model Accuracy:", accuracy_score(y_test, y_pred))

AdaBoost Classifier Model Accuracy: 0.9132475519948208


In [100]:
from sklearn.metrics import confusion_matrix
confusion_matrix(y_test,y_pred)

array([[10686,   283],
       [  789,   599]], dtype=int64)

In [101]:
ada = AdaBoostClassifier()
ada.fit(x_train, y_train)
y_train_ada = ada.predict(x_train)
y_hat_test_ada = ada.predict(x_test)

In [102]:
print(classification_report(y_train, y_train_ada))

              precision    recall  f1-score   support

          no       0.93      0.97      0.95     25579
         yes       0.67      0.39      0.49      3252

    accuracy                           0.91     28831
   macro avg       0.80      0.68      0.72     28831
weighted avg       0.90      0.91      0.90     28831



In [105]:
print(classification_report(y_test, y_hat_test_ada))

              precision    recall  f1-score   support

          no       0.93      0.97      0.95     10969
         yes       0.67      0.41      0.51      1388

    accuracy                           0.91     12357
   macro avg       0.80      0.69      0.73     12357
weighted avg       0.90      0.91      0.90     12357



# Inference from the models.

In [104]:
d=[['Logistic Regression',accuracy_test],['AdaBoost',accuracy_test_ada]]
df1 = pd.DataFrame(d, columns = ['Model','Test_Accuracy'])
print(df1)
   

                 Model  Test_Accuracy
0  Logistic Regression        0.91171
1             AdaBoost        0.91171


Both models ***Logistic Regression*** and  ***AdaBoost*** has the same accuracy. 

In [107]:
conf_matrix_test = confusion_matrix(y_test, y_hat_test)
print(conf_matrix_test)

[[10688   281]
 [  810   578]]


***The TruePostive rate for this Logistic Regression is 10688 and the true negative rate is also low when compared to Adaboost model.***